### PR TITLE
[Stablex] Better driver logging

### PR DESCRIPTION
**Pull request by fleupold**
Right now our log output is pretty messy and verbose. This PR addresses it in the following way:

1. No longer log every 5 seconds that we have already applied the current batch
2. Don't invoke the solver on empty auctions (leading to some complaints in the solver)
3. Don't submit solutions that don't have any volume (and thus also no utility). That would just be waisting gas.

On top of that we update the smart contracts to not show the annoying error message about a Set not being readable before there are any entries.

### Test Plan

Running the demo with the linear solver, will now show

```
stablex_1         |     Finished dev [unoptimized + debuginfo] target(s) in 0.23s
stablex_1         |      Running `target/debug/stablex`
stablex_1         | 2019-09-26 16:22:50,937 INFO  [driver::util] Using linear optimisation price finder
stablex_1         | 2019-09-26 16:22:50,988 INFO  [driver::driver::stablex_driver] No orders in batch 5231715
stablex_1         | 2019-09-26 16:22:50,988 INFO  [driver::driver::stablex_driver] Not submitting 0 volume solution for batch 5231715
stablex_1         | 2019-09-26 16:23:54,621 INFO  [driver::driver::stablex_driver] Computed solution: Solution { surplus: Some(3596998402997999639577936039039039039039), prices: [1000000000000000000, 2000000000000000000, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], executed_buy_amounts: [998999000999999899899, 1996000003997999800000], executed_sell_amounts: [1999997999999999799597, 998999000999999899899] }
stablex_1         | 2019-09-26 16:23:54,987 INFO  [driver::driver::stablex_driver] Successfully applied solution to batch 5231716
stablex_1         | 2019-09-26 16:24:02,981 INFO  [driver::driver::stablex_driver] Computed solution: Solution { surplus: Some(3596998403362023701801864760696680), prices: [1000000000000000000, 2000000000000000000, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], executed_buy_amounts: [998999001101101, 1996000004200000], executed_sell_amounts: [1999998000202404, 998999001101101] }
```

### CODE REVIEW

**Comment by bh2smith** [comment]
You read my mind! My only question here is what happens when no auction results are submitted? Is the batch just ignored and forgotten?

**Comment by bh2smith** [comment]
Still seeing this weird problem with arrayvec.

File: driver/src/driver/stablex_driver.rs [hunk 1]
```diff
@@ -25,17 +27,28 @@ impl<'a> StableXDriver<'a> {
...
+        };
 
-        info!("Submitting solution: {:?}", &solution);
-        self.contract.submit_solution(batch, orders, solution)?;
+        let submitted = if solution.executed_sell_amounts.iter().sum::<u128>() > 0 {
+            self.contract.submit_solution(batch, orders, solution)?;
+            info!("Successfully applied solution to batch {}", batch);
+            true
+        } else {
+            info!("Not submitting 0 volume solution for batch {}", batch);
```
Comment (bh2smith) [inline driver/src/driver/stablex_driver.rs]: Should (or could) this say something like "not submitting trivial solution"

Is a zero volume solution equivalent to trivial? I imagine it must be.

Comment (fleupold) [inline driver/src/driver/stablex_driver.rs]:
```suggestion
            info!("Not submitting trivial solution for batch {}", batch);
```

**Pull request merged**